src/msg/simple/Pipe.cc

```diff
@@ -2461,7 +2461,7 @@ int Pipe::tcp_read(char *buf, int len)
     buf += got;
     //lgeneric_dout(cct, DBL) << "tcp_read got " << got << ", " << len << " left" << dendl;
   }
-  return len;
+  return 0;
```

Comment:
yes, it seemed better for the length of read, but you can see the comment of tcp_read in   Pipe.h. 

    /**
     * do a blocking read of len bytes from socket
     *
     * @param buf buffer to read into
     * @param len exact number of bytes to read
     * @return 0 for success, or -1 on error
     */
    int tcp_read(char *buf, int len);

since both are ok, caller doesn't rely on the result code of tcp_read